### PR TITLE
RCOCOA-2070: Added sync mixed collection tests

### PR DESCRIPTION
**Pull request by dianaafanador3**
Collections in mixed sync tests

### CODE REVIEW

**Comment by dianaafanador3** [comment]
The failing tests correspond to new new workflow added for Xcode 16 beta, which needs Xcode cloud support for the beta and probably this [PR](https://github.com/realm/realm-swift/pull/8618) by @tgoyne

File: Realm/ObjectServerTests/SwiftCollectionSyncTests.swift [hunk 1]
```diff
@@ -338,3 +338,602 @@ class CollectionSyncTestCase: SwiftSyncTestCase {
...
+                // This happens when we download a fresh realm
+                XCTAssertEqual(object.anyCol.listValue?[index], .double(Double(value.floatValue!)))
+            }
+        } else if case .date = value {
+            // Date is not exact when synced
+            let date = object.anyCol.listValue?[index].dateValue
+            XCTAssertTrue(Calendar.current.isDate(date!, equalTo: value.dateValue!, toGranularity: .second))
+        } else {
+            XCTAssertEqual(object.anyCol.listValue?[index], value)
+        }
```
Comment (tgoyne) [inline Realm/ObjectServerTests/SwiftCollectionSyncTests.swift:406]: All of the checking logic here is duplicated three times and it should be pretty easy to extract.

File: Realm/ObjectServerTests/SwiftCollectionSyncTests.swift [hunk 2]
```diff
@@ -338,3 +338,602 @@ class CollectionSyncTestCase: SwiftSyncTestCase {
...
+        for (index, value) in list.enumerated() {
+            try await write { realm in
+                let object = SwiftTypesSyncObject()
+                object.anyCol = value
+                object.stringCol = "\(self.name)_\(index)"
+                realm.add(object)
+            }
+
+            let realm = try await openRealm()
+            let results0 =  try await realm.objects(SwiftTypesSyncObject.self).where { $0.stringCol == "\(self.name)_\(index)" }.subscribe()
```
Comment (tgoyne) [inline Realm/ObjectServerTests/SwiftCollectionSyncTests.swift:439]: None of the `.subscribe()`s in any of these tests do anything as the initial subscriptions are subscribing to all objects already.

File: Realm/ObjectServerTests/SwiftCollectionSyncTests.swift [hunk 3]
```diff
@@ -338,3 +338,602 @@ class CollectionSyncTestCase: SwiftSyncTestCase {
...
+            let date = object.anyCol.dictionaryValue?[key]?.dateValue
+            XCTAssertTrue(Calendar.current.isDate(date!, equalTo: value.dateValue!, toGranularity: .second))
+        } else {
+            XCTAssertEqual(object.anyCol.dictionaryValue?[key], value)
+        }
+    }
+
+    @MainActor func testSyncAnyRealmValue() async throws {
+        let list = sampleData
+        for (index, value) in list.enumerated() {
```
Comment (tgoyne) [inline Realm/ObjectServerTests/SwiftCollectionSyncTests.swift:430]: This test is extremely slow due to that it creates and synchronizes two Realms for each value to test. Creating all of the objects in a single Realm cuts the runtime from 8 seconds to .75 seconds on my machine.

File: Realm/ObjectServerTests/SwiftCollectionSyncTests.swift [hunk 4]
```diff
@@ -338,3 +338,602 @@ class CollectionSyncTestCase: SwiftSyncTestCase {
...
+        // Add initial list
+        try realm1.write {
+            let object = SwiftTypesSyncObject()
+            object.anyCol = AnyRealmValue.fromArray([])
+            object.stringCol = self.name
+            realm1.add(object)
+        }
+        try await realm1.syncSession?.wait(for: .upload)
+        try await realm2.syncSession?.wait(for: .download)
+        try await realm2.syncSession?.wait(for: .upload)
```
Comment (tgoyne) [inline Realm/ObjectServerTests/SwiftCollectionSyncTests.swift:612]: Waiting for upload completion should not be doing anything here. These tests do need to be calling `await realm2.asyncRefresh()` after waiting for `.download`, though.

File: ci_scripts/ci_post_clone.sh [hunk 5]
```diff
@@ -18,6 +18,7 @@ install_dependencies() {
         install_ruby
     elif [[ "$CI_WORKFLOW" == "sync"* ]]; then
     # elif [[ "$CI_WORKFLOW" == "sync"* ]] || [[ "$CI_WORKFLOW" == "swiftpm"* ]]; then
+        brew install node
```
Comment (tgoyne) [inline ci_scripts/ci_post_clone.sh:21]: setup-baas installs node if needed.

File: RealmSwift/Tests/MixedCollectionTest.swift [hunk 6]
```diff
@@ -986,4 +986,31 @@ class MixedCollectionTest: TestCase, @unchecked Sendable {
...
+            realm.add(o)
+
+        }
+        XCTAssertEqual(o.anyValue.value.dictionaryValue?["key1"], .string("hello"))
+        XCTAssertEqual(o.anyValue.value.dictionaryValue?["key2"], .bool(false))
+
+        try realm.write {
+            o.anyValue.value = AnyRealmValue.fromDictionary(dictionary1)
+        }
+        XCTAssertEqual(o.anyValue.value.dictionaryValue?["key1"], .string("adios"))
```
Comment (rorbech) [inline RealmSwift/Tests/MixedCollectionTest.swift:1014]: Maybe also add an assertion that `key2` is not present anymore. We had issues that we accidentally didn't clear an existing collection as part of the assignment during development in other SDKs.

File: Realm/ObjectServerTests/SwiftCollectionSyncTests.swift [hunk 7]
```diff
@@ -338,3 +338,602 @@ class CollectionSyncTestCase: SwiftSyncTestCase {
...
+            object.stringCol = self.name
+            realm.add(object)
+        }
+
+        let realm = try await openRealm()
+        let results0 =  try await realm.objects(SwiftTypesSyncObject.self).where { $0.stringCol == name }.subscribe()
+        XCTAssertEqual(results0.count, 1)
+
+        let o = results0.first!
+        XCTAssertEqual(o.anyCol.listValue?[1], .bool(false))
```
Comment (rorbech) [inline Realm/ObjectServerTests/SwiftCollectionSyncTests.swift:515]: Nitpick: Defining a local variable for `o.anyCol.listValue?` would make it a bit more readable.

File: Realm/ObjectServerTests/SwiftCollectionSyncTests.swift [hunk 8]
```diff
@@ -338,3 +338,602 @@ class CollectionSyncTestCase: SwiftSyncTestCase {
...
+            object.stringCol = self.name
+            realm.add(object)
+        }
+
+        let realm = try await openRealm()
+        let results0 =  try await realm.objects(SwiftTypesSyncObject.self).where { $0.stringCol == name }.subscribe()
+        XCTAssertEqual(results0.count, 1)
+
+        let o = results0.first!
+        XCTAssertEqual(o.anyCol.dictionaryValue?["key1"], .bool(false))
```
Comment (rorbech) [inline Realm/ObjectServerTests/SwiftCollectionSyncTests.swift:545]: Same. Using variables for common parts of the expressions would improve readability.

**Review by Naveen-C-Ramachandrappa — APPROVED** [review]

**Review by rorbech — APPROVED** [review]

**Comment by dianaafanador3** [comment]
@tgoyne solved all the comments, can you take a last look at this PR

**Review by tgoyne — APPROVED** [review]

**Pull request merged**